packages/neon-core/src/tx/transaction/main.ts

```diff
@@ -79,3 +117,31 @@ export function deserializeWitnesses(
   );
   return Object.assign(tx, { scripts });
 }
+
+export function formatSender(sender: string | undefined): string {
+  if (!sender) {
+    return "";
+  }
+  if (sender.length === 42 && sender.startsWith("0x")) {
+    const hex = sender.slice(2);
+    ensureHex(hex);
+    return hex;
+  } else if (sender.length === 40) {
+    ensureHex(sender);
+    return sender;
+  } else if (sender.length === 34) {
+    return getScriptHashFromAddress(sender);
+  } else {
+    throw new Error(`Sender format error: ${sender}`);
+  }
+}
+
+export function getVarSize(value: number): number {

```

Comment:
This looks suspiciously like a util method. Have we changed how we read var bytes?

Reply:
This is another method to calculate the compressed size according to number value. It's used in calculating networkFee according to tx size. The core code use this method name, maybe we can change it to another name to avoid confusion?

Reply:
yes. And why are we exporting this? Where else will we use it other than in Transaction?

Reply:
It's only used in Transaction. Since only one type of transaction lingers, most functions in main.ts is only used in Transaction. But moving all these methods to Transaction would make a huge class. Anyway these are not exported from module 'tx', which I think is good enough.